tests/unit/job/test_query.py

```diff
@@ -989,6 +989,40 @@ def test_result(self):
             [query_results_call, query_results_call, reload_call, query_page_call]
         )
 
+    def test_result_dry_run(self):
+        query_resource_done = {
+            "jobComplete": True,
+            "jobReference": {
+                "projectId": self.PROJECT,
+                "jobId": self.JOB_ID,
```

Comment:
Dry run queries don't actually get a job ID.

I got annoyed by some of our validation in https://github.com/googleapis/python-bigquery/pull/1014 and removed it because of that...